app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -739,6 +739,8 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
 
   $scope.updateAuthStatus = function(updatedValue) {
     $scope.angularForm[$scope.currentTab + '_auth_status'].$setViewValue(updatedValue);
+    $scope.angularForm.$valid = updatedValue;
+    $scope.angularForm.$invalid = !updatedValue;
```

Comment:
I would leave this function alone. Nothing to be done here.
We should probably examine the haml

Reply:
Do you know that your [previous PR](https://github.com/ManageIQ/manageiq-ui-classic/pull/3958) with the field you added was buggy and didn't enable "save" button on change? This change solves that!

Reply:
Not sure what I am missing here, I checked and checked again but Save button is enabled for me

Reply:
Let the form organically come to the conclusion if it's valid or invalid, that is why we have so many field-level validations in place.
Modifying`$valid` / `$invalid` is not recommended, we don't know what implications it would have in other use cases.

Reply:
`updateAuthStatus` called in the same file from two places of the `validateClicked` function:
- https://github.com/ManageIQ/manageiq-ui-classic/blob/fc2600c927324e260bbd3be9d975fc5d7e4ba40d/app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js#L725
- https://github.com/ManageIQ/manageiq-ui-classic/blob/fc2600c927324e260bbd3be9d975fc5d7e4ba40d/app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js#L727
Issue was that even when the validation succeed, form didn't activated "Add/Save" button.

Reply:
> Not sure what I am missing here, I checked and checked again but Save button is enabled for me (on the master branch)

It got enabled after you edit "Endpoint URL" field for the AWS provider? When you change existing one?
In this branch - will it be enabled when you edit "Assume role ARN" field without those two lines? (try to comment them).

Reply:
Ugh, sorry @AlexanderZagaynov but this is really not the right fix.

If you need to set `$valid` or `$invalid` manually, it probably means you're doing something outside the angular change detection cycle, otherwise, the form validation should kick and set the valid state of the form.

You need to look into why angularForm is not valid, there should be an `$errors` field which will tell you which validation actually failed.